chapters/08-documentation.Rmd

```diff
@@ -4,7 +4,49 @@
 - Function documentation (with roxygen2), part 2
 - Running spell checks and styling (with spelling and styler)
 
+## Learning Objectives
 
+* Describe the purpose of documenting code and packages
+* Document functions with `roxygen`
+* Create a README and vignette for package documentation
+* Ensure that documentation is consistent with up-to-date style guidelines using `styler`
+
+## Outline
+
+* Introduction: the purpose of documentation
+
+  * Text: documentation is for collaborators, users, and future you. It is key for openness, reproducibility, and sustainability.
+
+* Documentation workflow: documenting the small stuff ([R Packages reference](https://r-pkgs.org/man.html?q=roxygen#man))
+
+  * Text: why document functions
+  * Code: function documentation cycle. Add `roxygen` comments (with cursor in function definition, do `Code > Insert roxygen skeleton` in RStudio). Run `devtools::document()`.
```

Comment:
Are we going to be teaching RStudio shortcuts? I don't have an opinion, but if we are I think we should keep track of which ones we are showing.

Reply:
Agreed - I don't think there are currently shortcuts in what I wrote, but I could be wrong?

Reply:
That's a really good idea!! We could actually track them in data file and refer to them with inline R code.